node/src/client/mod.rs

```diff
@@ -39,17 +39,30 @@ use snarkvm::{
         store::ConsensusStorage,
         Ledger,
     },
+    prelude::block::Transaction,
 };
 
 use aleo_std::StorageMode;
 use anyhow::Result;
 use core::future::Future;
+use lru::LruCache;
 use parking_lot::Mutex;
 use std::{
     net::SocketAddr,
-    sync::{atomic::AtomicBool, Arc},
+    num::NonZeroUsize,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize, Ordering::Relaxed},
+        Arc,
+    },
+    time::Duration,
 };
-use tokio::task::JoinHandle;
+use tokio::{task::JoinHandle, time::sleep};
+
+const VERIFICATION_CONCURRENCY_LIMIT: usize = 6; // 8 deployments of MAX_NUM_CONSTRAINTS will run out of memory.
```

Comment:
Should this be dynamic based on your machine? 

How much memory was this estimation based on?

Reply:
This was based on the `Clients: 16GB of memory` requirement in the snarkOS README. Please confirm if you want me to make it a dynamic value from a cli flag. 

Sidenote: given our syncing issues I understood the CPU/memory requirements need to go up anyway.